exercises/takudzwa_makoni/ex02_oo_basics/src/circle.h

```diff
@@ -0,0 +1,31 @@
+#ifndef CIRCLE_H
+#define CIRCLE_H
+
+class circle : public shape { // circle class
+
+private:
+
+	
+	const std::string name;
+	const int numOfSides;
+	const double pi = 3.14159;
+
+public:
+
+	double radius;
+
+	virtual int getNumOfSides() const override { return numOfSides; }
+	virtual std::string getName() const override { return name; }
+
+	virtual double getArea() const override { return pi * radius * radius; }
+	virtual double getPerimeter() const override { return pi * 2 * radius; }
+
+	//constructor
+	circle::circle() : name("circle"), numOfSides(1) {
```

Comment:
`circle::circle()` ... `rectangle::rectangle` are causing compiler errors on RHEL7. We discussed already.

Reply:
I want to set the radius of my circle. I think that's a reasonable thing to do. Please change the constructor to allow the variable parts of the type to be "injected" so constructor should look like `circle(double rad) : radius(rad) {...}` or similar

Reply:
i've made radius in circle class private instead of public, and written get/set member functions for all shapes. circle::circle() changed to circle() etc. ive put default values in the initialiser list.